### ISSUE TITLE
Value dependent typing signature with self-defined generic types incorrectly handled

### ISSUE THREAD

**Issue by Blealtan** [microsoft/pylance-release#2784]
## Environment data

-   Language Server version: `Pylance language server 2022.4.3 (pyright 88f2a424)`
-   OS and version: `Linux 5.10.0-13-amd64 #1 SMP Debian 5.10.106-1 (2022-03-17) x86_64 GNU/Linux`
-   Python version (& distribution if applicable, e.g. Anaconda): `Python 3.8.12`

## Expected behavior

Self-defined generic types should allow values to appear in their arguments. By "self-defined" I mean anything not in the standard `typing` module, including simplest things like `RenamedLiteral = typing.Literal`.

## Actual behavior

Shown as below.

![image](https://user-images.githubusercontent.com/6376623/167290382-c9e12542-b0ff-4f33-a230-cf0a1345de45.png)

Notice that the `correct` function is correctly handled, but `wrong` is not; the only difference is one uses the standard `Literal` and the other use something else. I've tested other self-defined generics and also failed.

Besides that, `highlighted` has the `"input"` string literal highlighted just like a function name. Maybe it's a helpful clue so I include it here.

## Logs

```
[Info  - 5:32:08 PM] (130925) Pylance language server 2022.4.3 (pyright 88f2a424) starting
[Info  - 5:32:08 PM] (130925) Server root directory: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist
[Info  - 5:32:08 PM] (130925) No pyproject.toml file found.
[Info  - 5:32:08 PM] (130925) Setting pythonPath for service "<default>": "/bin/python3"
[Warn  - 5:32:08 PM] (130925) stubPath typings is not a valid directory.
[Info  - 5:32:08 PM] (130925) Assuming Python version 3.9
[Info  - 5:32:08 PM] (130925) Assuming Python platform Linux
[Info  - 5:32:08 PM] (130925) Searching for source files
[Info  - 5:32:08 PM] (130925) No source files found.
[Info  - 5:32:08 PM] (130925) Background analysis(1) root directory: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist
[Info  - 5:32:08 PM] (130925) Background analysis(1) started
[Info  - 5:32:32 PM] (130925) No pyproject.toml file found.
[Info  - 5:32:32 PM] (130925) Setting pythonPath for service "<default>": "/bin/python3"
[Warn  - 5:32:32 PM] (130925) stubPath typings is not a valid directory.
[Info  - 5:32:32 PM] (130925) Assuming Python version 3.9
[Info  - 5:32:32 PM] (130925) Assuming Python platform Linux
[Info  - 5:32:32 PM] (130925) Search paths for <default>
[Info  - 5:32:32 PM] (130925)   /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib
[Info  - 5:32:32 PM] (130925)   /home/spack/spack/opt/spack/linux-debian11-zen2/gcc-10.2.1/py-setuptools-58.2.0-gciep2xatzq6erdpqacj4adfli4v6weu/lib/python3.8/site-packages
[Info  - 5:32:32 PM] (130925)   /home/spack/spack/opt/spack/linux-debian11-zen2/gcc-10.2.1/py-pip-21.1.2-ekpecot777hn7awa4yhnqqhriemqyqaa/lib/python3.8/site-packages
[Info  - 5:32:32 PM] (130925)   /home/spack/spack/opt/spack/linux-debian11-zen2/gcc-10.2.1/py-setuptools-58.2.0-gciep2xatzq6erdpqacj4adfli4v6weu/lib/python3.8/site-packages
[Info  - 5:32:32 PM] (130925)   typings
[Info  - 5:32:32 PM] (130925)   /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stubs/...
[Info  - 5:32:32 PM] (130925)   /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/bundled/stubs
[Info  - 5:32:32 PM] (130925)   /home/spack/spack/opt/spack/linux-debian11-zen2/gcc-10.2.1/py-setuptools-58.2.0-gciep2xatzq6erdpqacj4adfli4v6weu/lib/python3.8/site-packages
[Info  - 5:32:32 PM] (130925)   /home/spack/spack/opt/spack/linux-debian11-zen2/gcc-10.2.1/py-pip-21.1.2-ekpecot777hn7awa4yhnqqhriemqyqaa/lib/python3.8/site-packages
[Info  - 5:32:32 PM] (130925)   /usr/lib/python3.9
[Info  - 5:32:32 PM] (130925)   /usr/lib/python3.9/lib-dynload
[Info  - 5:32:32 PM] (130925)   /home/blealtancao/.local/lib/python3.9/site-packages
[Info  - 5:32:32 PM] (130925)   /usr/local/lib/python3.9/dist-packages
[Info  - 5:32:32 PM] (130925)   /usr/lib/python3/dist-packages
[Info  - 5:32:32 PM] (130925) Adding fs watcher for library directories:
 /home/spack/spack/opt/spack/linux-debian11-zen2/gcc-10.2.1/py-setuptools-58.2.0-gciep2xatzq6erdpqacj4adfli4v6weu/lib/python3.8/site-packages
/home/spack/spack/opt/spack/linux-debian11-zen2/gcc-10.2.1/py-pip-21.1.2-ekpecot777hn7awa4yhnqqhriemqyqaa/lib/python3.8/site-packages
/usr/lib/python3.9
/usr/lib/python3.9/lib-dynload
/home/blealtancao/.local/lib/python3.9/site-packages
/usr/local/lib/python3.9/dist-packages
/usr/lib/python3/dist-packages
[Info  - 5:32:32 PM] (130925) Searching for source files
[Info  - 5:32:32 PM] (130925) No source files found.
(130925) [IDX(FG)] index libraries  (index) ...
(130925) [IDX(FG)]   read stdlib indices (72ms)
(130925) [IDX(FG)] index libraries  (index) [succeed] (73ms)
(130925) Background analysis message: setConfigOptions
(130925) Background analysis message: setImportResolver
(130925) Background analysis message: ensurePartialStubPackages
(130925) Background analysis message: setTrackedFiles
(130925) Background analysis message: markAllFilesDirty
(130925) Background analysis message: analyze
(130925) [BG(1)] analyzing: /home/blealtancao/test.py ...
(130925) [BG(1)]   parsing: /home/blealtancao/test.py (26ms)
(130925) [BG(1)]   parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 1ms] (124ms)
(130925) [BG(1)]   binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/builtins.pyi (26ms)
(130925) [BG(1)]   binding: /home/blealtancao/test.py (0ms)
(130925) [BG(1)]   checking: /home/blealtancao/test.py ...
(130925) [BG(1)]     parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 1ms] (12ms)
(130925) [BG(1)]     binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing_extensions.pyi (2ms)
(130925) [BG(1)]     parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing.pyi [fs read 0ms] (35ms)
(130925) [BG(1)]     binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing.pyi (11ms)
(130925) [BG(1)]     parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 0ms] (11ms)
(130925) [BG(1)]     binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (1ms)
(130925) [BG(1)]     parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/abc.pyi [fs read 0ms] (1ms)
(130925) [BG(1)]     binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/abc.pyi (0ms)
(130925) [BG(1)]   checking: /home/blealtancao/test.py (82ms)
(130925) [BG(1)] analyzing: /home/blealtancao/test.py (258ms)
(130925) Background analysis message: getSemanticTokens delta
(130925) [BG(1)] getSemanticTokens delta previousResultId:1652002329113 at /home/blealtancao/test.py (1ms)
(130925) Background analysis message: resumeAnalysis
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: getDiagnosticsForRange
(130925) [FG] parsing: /home/blealtancao/test.py (34ms)
(130925) [FG] parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 0ms] (127ms)
(130925) [FG] binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/builtins.pyi (26ms)
(130925) [FG] binding: /home/blealtancao/test.py (1ms)
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: setFileClosed
(130925) Background analysis message: analyze
(130925) Background analysis message: setFileOpened
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: getDiagnosticsForRange
(130925) [FG] parsing: /home/blealtancao/test.py (6ms)
(130925) [FG] parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 2ms] (79ms)
(130925) [FG] binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/builtins.pyi (27ms)
(130925) [FG] binding: /home/blealtancao/test.py (0ms)
(130925) Background analysis message: getSemanticTokens full
(130925) [BG(1)] getSemanticTokens full at /home/blealtancao/test.py ...
(130925) [BG(1)]   parsing: /home/blealtancao/test.py (1ms)
(130925) [BG(1)]   parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 1ms] (55ms)
(130925) [BG(1)]   binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/builtins.pyi (27ms)
(130925) [BG(1)]   binding: /home/blealtancao/test.py (0ms)
(130925) [BG(1)]   parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing.pyi [fs read 0ms] (17ms)
(130925) [BG(1)]   binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing.pyi (5ms)
(130925) [BG(1)]   parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 0ms] (3ms)
(130925) [BG(1)]   binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing_extensions.pyi (1ms)
(130925) [BG(1)] getSemanticTokens full at /home/blealtancao/test.py (121ms)
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: analyze
(130925) [BG(1)] analyzing: /home/blealtancao/test.py ...
(130925) [BG(1)]   checking: /home/blealtancao/test.py (2ms)
(130925) [BG(1)] analyzing: /home/blealtancao/test.py (2ms)
(130925) Background analysis message: resumeAnalysis
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: setFileOpened
(130925) Background analysis message: markFilesDirty
(130925) Background analysis message: setFileOpened
(130925) Background analysis message: markFilesDirty
(130925) [FG] parsing: /home/blealtancao/test.py (3ms)
(130925) [FG] binding: /home/blealtancao/test.py (1ms)
(130925) Background analysis message: analyze
(130925) [BG(1)] analyzing: /home/blealtancao/test.py ...
(130925) [BG(1)]   parsing: /home/blealtancao/test.py (1ms)
(130925) [BG(1)]   binding: /home/blealtancao/test.py (0ms)
(130925) [BG(1)]   checking: /home/blealtancao/test.py ...
(130925) [BG(1)]     parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 1ms] (4ms)
(130925) [BG(1)]     binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (6ms)
(130925) [BG(1)]     parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/abc.pyi [fs read 0ms] (1ms)
(130925) [BG(1)]     binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/abc.pyi (1ms)
(130925) [BG(1)]   checking: /home/blealtancao/test.py (25ms)
(130925) [BG(1)] analyzing: /home/blealtancao/test.py (26ms)
(130925) Background analysis message: resumeAnalysis
(130925) Background analysis message: getSemanticTokens delta
(130925) [BG(1)] getSemanticTokens delta previousResultId:1652002359374 at /home/blealtancao/test.py (1ms)
(130925) Background analysis message: setFileOpened
(130925) Background analysis message: markFilesDirty
(130925) [FG] parsing: /home/blealtancao/test.py (1ms)
(130925) [FG] binding: /home/blealtancao/test.py (0ms)
(130925) Background analysis message: analyze
(130925) [BG(1)] analyzing: /home/blealtancao/test.py ...
(130925) [BG(1)]   parsing: /home/blealtancao/test.py (0ms)
(130925) [BG(1)]   binding: /home/blealtancao/test.py (0ms)
(130925) [BG(1)]   checking: /home/blealtancao/test.py (12ms)
(130925) [BG(1)] analyzing: /home/blealtancao/test.py (13ms)
(130925) Background analysis message: resumeAnalysis
(130925) Background analysis message: getSemanticTokens delta
(130925) [BG(1)] getSemanticTokens delta previousResultId:1652002571525 at /home/blealtancao/test.py (1ms)
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: getDiagnosticsForRange
(130925) [FG] parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 1ms] (10ms)
(130925) [FG] binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing_extensions.pyi (3ms)
(130925) [FG] parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing.pyi [fs read 0ms] (32ms)
(130925) [FG] binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/typing.pyi (25ms)
(130925) [FG] parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 0ms] (18ms)
(130925) [FG] binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (5ms)
(130925) [FG] parsing: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/abc.pyi [fs read 0ms] (2ms)
(130925) [FG] binding: /home/blealtancao/.vscode-server/extensions/ms-python.vscode-pylance-2022.4.3/dist/typeshed-fallback/stdlib/abc.pyi (0ms)
(130925) Background analysis message: getDiagnosticsForRange
(130925) Background analysis message: getDiagnosticsForRange
```

## Code Snippet / Additional information

```python
from typing import Literal

def correct(x: Literal["/y"]):
    pass

RenamedLiteral = Literal

def wrong(x: RenamedLiteral["/y"]):
    pass

def highlighted(x: RenamedLiteral["input"]):
    pass
```

**Comment by Blealtan** [comment]
Then by running mypy I found the code is actually illegally typed, but in a slightly different way than what Pylance say:

```
test.py:8: error: Variable "test.RenamedLiteral" is not valid as a type
test.py:8: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
test.py:8: error: Invalid type comment or annotation
test.py:11: error: Variable "test.RenamedLiteral" is not valid as a type
test.py:11: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
test.py:11: error: Function "builtins.input" is not valid as a type
test.py:11: note: Perhaps you need "Callable[...]" or a callback protocol?
Found 4 errors in 1 file (checked 1 source file)
```

**Comment by erictraut** [comment]
Until very recently, pyright (the type checker that underlies pylance) didn't support aliasing of `Literal`. With [this change](https://github.com/microsoft/pyright/issues/3394), this now works. It will be included in the next release of pylance.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.5.1, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202251-11-May-2022)
